src/PolynomialTransform.cc

```diff
@@ -0,0 +1,280 @@
+// -*- LSST-C++ -*-
+
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST/AURA
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include "lsst/meas/astrom/PolynomialTransform.h"
+#include "lsst/meas/astrom/SipTransform.h"
+#include "lsst/meas/astrom/detail/polynomialUtils.h"
+
+namespace lsst { namespace meas { namespace astrom {
+
+PolynomialTransform PolynomialTransform::convert(ScaledPolynomialTransform const & scaled) {
+    return compose(
+        scaled.getOutputScalingInverse(),
+        compose(scaled.getPoly(), scaled.getInputScaling())
+    );
+}
+
+PolynomialTransform PolynomialTransform::convert(SipForwardTransform const & other) {
+    PolynomialTransform poly = other.getPoly();
+    poly._xCoeffs(1, 0) += 1;
+    poly._yCoeffs(0, 1) += 1;
```

Comment:
Done.  It's to account for the extra terms outside the sum in the SIP definition formula; you can fold those into the sum by adding one to these two coefficients.